src/unistd.rs

```diff
@@ -2729,3 +2751,42 @@ impl Group {
         })
     }
 }
+
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::{fs, thread, io};
+
+    /// Tests setting the filesystem UID with `setfsuid`.
+    #[test]
+    fn test_setfsuid() {
+        use std::os::unix::fs::PermissionsExt;
+
+        // get the UID of the "nobody" user
+        let nobody = User::from_name("nobody").unwrap().unwrap();
+
+        // create a temporary file with permissions '-rw-r-----'
+        let file = tempfile::NamedTempFile::new().unwrap();
+        let temp_path = file.into_temp_path();
+        let temp_path_2 = (&temp_path).to_path_buf();
+        let mut permissions = fs::metadata(&temp_path).unwrap().permissions();
+        permissions.set_mode(640);
+
+        // spawn a new thread where to test setfsuid
+        thread::spawn(move || {
+            // set filesystem UID
+            let _ = setfsuid(nobody.uid);
```

Comment:
To determine success, you should follow the procedure described in the man page.  Call `setfsuid(-1)` and check whether the fsuid was actually changed.  I suspect that when running on Travis, it won't be.

http://man7.org/linux/man-pages/man2/setfsuid.2.html

Reply:
I added an assertion for the fsuid change.